contracts/voting-verifier/src/execute.rs

```diff
@@ -1,32 +1,37 @@
-use axelar_wasm_std::operators::Operators;
 use cosmwasm_std::{
     to_binary, Deps, DepsMut, Env, MessageInfo, QueryRequest, Response, Storage, WasmMsg, WasmQuery,
 };
 
-use axelar_wasm_std::voting::{PollId, Vote};
-use axelar_wasm_std::{nonempty, snapshot, voting::WeightedPoll};
+use axelar_wasm_std::{
+    nonempty,
+    operators::Operators,
+    snapshot,
+    voting::{PollId, Vote, WeightedPoll},
+    VerificationStatus,
+};
+
 use connection_router::state::{ChainName, Message};
 use service_registry::msg::QueryMsg;
 use service_registry::state::Worker;
 
-use crate::error::ContractError;
 use crate::events::{
     PollEnded, PollMetadata, PollStarted, TxEventConfirmation, Voted, WorkerSetConfirmation,
 };
 use crate::msg::{EndPollResponse, VerifyMessagesResponse};
-use crate::query::{
-    is_verified, is_worker_set_verified, msg_verification_status, VerificationStatus,
-};
+use crate::query::worker_set_status;
 use crate::state::{self, Poll, PollContent, POLL_MESSAGES, POLL_WORKER_SETS};
 use crate::state::{CONFIG, POLLS, POLL_ID};
+use crate::{error::ContractError, query::message_status};
 
 pub fn verify_worker_set(
     deps: DepsMut,
     env: Env,
     message_id: nonempty::String,
     new_operators: Operators,
 ) -> Result<Response, ContractError> {
-    if is_worker_set_verified(deps.as_ref(), &new_operators)? {
+    let status = worker_set_status(deps.as_ref(), &new_operators)?;
+    if status == VerificationStatus::SucceededOnChain || status == VerificationStatus::FailedOnChain
```

Comment:
I personally would love to see this to be extracted into a method of the enum, i.e. `is_xxx` something like this, I do not have a good name for it though.